bin/host/src/server.rs

```diff
@@ -20,6 +19,16 @@ pub struct PreimageServer<P, H, B> {
     backend: Arc<B>,
 }
 
+#[derive(Debug, thiserror::Error)]
+enum PreimageServerError {
+    #[error("Failed to serve preimage request: {0}")]
+    PreimageRequestFailed(PreimageOracleError),
+    #[error("Failed to route hint: {0}")]
+    RouteHintFailed(PreimageOracleError),
+    #[error("Join error: {0}")]
+    ExecutionError(#[from] tokio::task::JoinError),
+}
```

Comment:
this is used it pub api and should also be pub

a one-line doc also wouldn't hurt here

Reply:
Agree with @mattsse here